### ISSUE TITLE
Can OpenCL caffe be built on windows?

### ISSUE THREAD

**Issue by Chris-R-R** [naibaf7/caffe#65]
Hi,
I noticed that there is a separate branch of Caffe for windows support but what are my options if I want both Windows support and OpenCL support?

Are there any benchmarks on how OpenCL compares to CUDA speedwise? Assuming a GPU that supports both, and on image classification tasks.

**Comment by naibaf7** [comment]
There are currently no up-to-date benchmarks that compare the two. It also depends a lot on what GPU you have and which network is used.

As a rule of thumb, an AMD GPU with X TFLOPS (using LibDNN) will be half as fast as an nVidia GPU with X TFLOPS (using cuDNN) at the moment, which is still very significantly more than CPU performance.

Yes, OpenCL Caffe does support Windows. Just read both the instructions from OpenCL and Windows Caffe in the Readme. You'll need MSVC, ViennaCL and Caffe to be able to compile it on Windows, it is not very difficult.

**Comment by psyhtest** [comment]
Feel free to use [Windows binaries](https://github.com/dividiti/ck-caffe/wiki/Installation#Packaging_Caffe_library_and_binaries_for_distribution) prebuilt via [Collective Knowledge](http://cknowledge.org/ai).

**Comment by psyhtest** [comment]
@Chris-R-R We have recently released benchmarking data for the NVIDIA GTX 1080 across 4 models and 14 variants of Caffe which can be viewed [here](https://nbviewer.jupyter.org/github/dividiti/ck-caffe-nvidia-gtx1080/blob/master/script/analysis/ck-caffe-nvidia-gtx1080.20170518.ipynb?raw). While the data is about 6 months old, the community can easily perform similar experiments across other devices, variants and models.  Hope this helps!

**Comment by Chris-R-R** [comment]
Hi I am looking into this again - I need to build OpenCL Caffe with VS2017. Unfortunately the build_win.cmd does not seem to support this but I was able to build the windows branch with VS2015 and update the VS solution/project to VS2017 and get that to build. I am now using this solution as a starting point to try and get the OpenCL branch to build with VS2017. So far there are a few problems:

* common.hpp references caffe/macros.hpp and caffe/export.hpp - these files are missing for me.

* blob.hpp references a DataType type (return value from data_type() function). This doesn't seem to be defined anywhere, nor the CAFFE_FLOAT,CAFFE_DOUBLE,CAFFE_HALF etc.


I used Git to fetch the code.

**Comment by naibaf7** [comment]
That's right, the latest OpenCL branch does not support Windows yet - update will follow this fall.

**Comment by Chris-R-R** [comment]
Thanks. Any idea how far back I have to go to get a version that does work on windows? If I can fetch a particular revision?

**Comment by naibaf7** [comment]
35a53ee98e033bfd95aabc0cb831ce9160fe196a probably
